### PR TITLE
test: Add tests for update mobile number popover and my-profile page

### PR DESCRIPTION
**Pull request by yksolanki9**
Clickup 
- https://app.clickup.com/t/85zt2xar4
- https://app.clickup.com/t/85zt2xb7y

Code coverage:
Screenshot 2023-05-23 at 6 52 49 PM

Covered only newly added methods in profile page:
Screenshot 2023-05-23 at 9 26 54 PM

### CODE REVIEW

File: src/app/fyle/my-profile/my-profile.page.spec.ts [hunk 1]
```diff
@@ -121,80 +136,35 @@ xdescribe('MyProfilePage', () => {
...
-          title: 'Edit Mobile Number',
-          ctaText: 'Save',
-          inputLabel: 'Mobile Number',
-          inputValue: apiEouRes.ou.mobile,
-          inputType: 'tel',
-          isRequired: false,
+  describe('setInfoCardsData(): ', () => {
+    let allInfoCardsData = [];
+    beforeAll(() => {
+      allInfoCardsData = [
```
Comment (Dimple16) [inline src/app/fyle/my-profile/my-profile.page.spec.ts:142]: does `allInfoCardsData` have a model? haven't checked the previous PRs if any for context, 
if yes, let us add the model and move this data to mock data file

File: src/app/fyle/my-profile/update-mobile-number/update-mobile-number.component.spec.ts [hunk 2]
```diff
@@ -1,24 +1,197 @@
...
+  it('closePopover(): should dismiss the modal', () => {
+    popoverController.dismiss.and.resolveTo();
+    const closeCta = getElementBySelector(fixture, 'ion-button') as HTMLIonButtonElement;
+    click(closeCta);
+
+    expect(popoverController.dismiss).toHaveBeenCalledTimes(1);
+  });
+
+  it('validateInput(): should set error message if input is invalid', () => {
+    const valueErrorMapping = [
```
Comment (Dimple16) [inline src/app/fyle/my-profile/update-mobile-number/update-mobile-number.component.spec.ts:91]: move this to mock data and add a model

**Review by Dimple16 — CHANGES_REQUESTED** [review]
added comments

File: src/app/fyle/my-profile/my-profile.page.spec.ts (hunk 1 above)
Comment (yksolanki9) [reply src/app/fyle/my-profile/my-profile.page.spec.ts]: Created a model and moved data to a data file

File: src/app/core/mock-data/value-error-mapping-for-update-mobile-number-popover.data.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
+interface ValueErrorMapping {
```
Comment (yksolanki9) [inline src/app/core/mock-data/value-error-mapping-for-update-mobile-number-popover.data.ts:1]: This array just contains the expected input/output for tests and is not being used anywhere in ts code, so I've created the model inside the data file itself

**Review by Dimple16 — APPROVED** [review]

**Pull request merged**